Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/day/sheets/location/ContactDiaryLocationBottomSheetDialogFragment.kt

```diff
@@ -28,18 +34,40 @@ class ContactDiaryLocationBottomSheetDialogFragment : BottomSheetDialogFragment(
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        binding.contactDiaryLocationBottomSheetCloseButton.buttonIcon.setOnClickListener {
-            viewModel.closePressed()
+        val location = navArgs.selectedLocation
+        if (location != null) {
+            binding.contactDiaryLocationBottomSheetTextInputEditText.setText(location.locationName)
+            binding.contactDiaryLocationBottomSheetDeleteButton.visibility = View.VISIBLE
+            binding.contactDiaryLocationBottomSheetDeleteButton.setOnClickListener {
+                DialogHelper.showDialog(deleteLocationConfirmationDialog)
+            }
+            binding.contactDiaryLocationBottomSheetSaveButton.setOnClickListener {
+                viewModel.updateLocation(location)
+            }
+        } else {
+            binding.contactDiaryLocationBottomSheetDeleteButton.visibility = View.GONE
+            binding.contactDiaryLocationBottomSheetSaveButton.setOnClickListener {
+                viewModel.addLocation()
+            }
         }
 
-        binding.contactDiaryLocationBottomSheetSaveButton.setOnClickListener {
-            viewModel.saveLocation()
+        binding.contactDiaryLocationBottomSheetCloseButton.buttonIcon.setOnClickListener {
+            viewModel.closePressed()
         }
 
         binding.contactDiaryLocationBottomSheetTextInputEditText.doAfterTextChanged {
             viewModel.textChanged(it.toString())
         }
 
+        binding.contactDiaryLocationBottomSheetTextInputEditText.setOnEditorActionListener { v, actionId, event ->
+            return@setOnEditorActionListener when (actionId) {
+                EditorInfo.IME_ACTION_DONE -> {
+                    false
+                }
+                else -> true
+            }
+        }
```

Comment:
```suggestion
        binding.contactDiaryLocationBottomSheetTextInputEditText.setOnEditorActionListener { v, actionId, event ->
            when (actionId) {
                EditorInfo.IME_ACTION_DONE -> false
                else -> true
            }
        }
```
or just
```suggestion
        binding.contactDiaryLocationBottomSheetTextInputEditText.setOnEditorActionListener { v, actionId, event ->
            actionId != EditorInfo.IME_ACTION_DONE
        }
```
although it's a lot less readable

Reply:
I removed the brackets

Reply:
the return is also unnecessary